nanochat/checkpoint_manager.py

```diff
@@ -110,27 +110,27 @@ def find_largest_model(checkpoints_dir):
         candidates.sort(key=lambda x: x[0], reverse=True)
         return candidates[0][1]
     # 2) if that failed, take the most recently updated model:
-    model_tags.sort(key=lambda x: os.path.getmtime(os.path.join(checkpoints_dir, x)), reverse=True)
+    model_tags.sort(key=lambda x: (checkpoints_dir / x).stat().st_mtime, reverse=True)
     return model_tags[0]
 
 
-def find_last_step(checkpoint_dir):
+def find_last_step(checkpoint_dir: Path):
     # Look into checkpoint_dir and find model_<step>.pt with the highest step
-    checkpoint_files = glob.glob(os.path.join(checkpoint_dir, "model_*.pt"))
+    checkpoint_files = list(checkpoint_dir.glob("model_*.pt"))
     if not checkpoint_files:
         raise FileNotFoundError(f"No checkpoints found in {checkpoint_dir}")
-    last_step = int(max(os.path.basename(f).split("_")[-1].split(".")[0] for f in checkpoint_files))
+    last_step = int(max(f.name.split("_")[-1].split(".")[0] for f in checkpoint_files))
```

Comment:
By using `stem` I believe we can drop the splitting on `"."` entirely:
```suggestion
    last_step = int(max(f.stem.split("_")[-1] for f in checkpoint_files))
```

Reply:
There shouldn't be more than one suffix, so it should be fine